tests/test.rs

```diff
@@ -45,7 +45,7 @@ impl Token {
     }
 
     fn approve(&mut self, from: &Keypair, spender: &Identifier, amount: BigInt) {
-        let from_id = Identifier::Ed25519(from.public.to_bytes());
+        let from_id = Identifier::Ed25519(from.public.to_bytes().try_into_val(&self.0).unwrap());
```

Comment:
If we make the ArrayBinary size type `usize` instead of `u32` this line can become:
```suggestion
        let from_id = Identifier::Ed25519(from.public.to_bytes().into_val(&self.0));
```